lib/Server.js

```diff
@@ -292,14 +290,12 @@ class Server {
      * ]
      */
     this.options.proxy.forEach((proxyConfigOrCallback) => {
-      let proxyConfig;
       let proxyMiddleware;
 
-      if (typeof proxyConfigOrCallback === 'function') {
-        proxyConfig = proxyConfigOrCallback();
-      } else {
-        proxyConfig = proxyConfigOrCallback;
-      }
+      let proxyConfig =
```

Comment:
maybe const?

Reply:
Actually I think let is needed, since `proxyConfig` is set to something else later under certain conditions. Probably why tests are failing.